node/src/blockchain/blockchain_interface.rs

```diff
@@ -1140,162 +1292,412 @@ mod tests {
-                169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-                0, 0, 0, 98, 108, 97, 104, 49, 50, 51, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 45, 121, 136, 61, 32, 0,
+                169, 5, 156, 187, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                99, 114, 101, 100, 105, 116, 111, 114, 49, 50, 51, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 156, 231, 56, 4,
             ]),
             chain_id: Some(1),
         };
-        let consuming_wallet_secret_raw_bytes = b"my-wallet+++++++++++++++++++++++";
-        let secret = (&Bip32ECKeyProvider::from_raw_secret(consuming_wallet_secret_raw_bytes)
-            .unwrap())
-            .into();
-        let signed_transaction = subject
+        let second_signed_transaction = subject
             .web3
             .accounts()
-            .sign_transaction(transaction_parameters_expected.clone(), &secret)
+            .sign_transaction(second_tx_parameters.clone(), &secret_key)
             .wait()
             .unwrap();
-        let hash = signed_transaction.transaction_hash;
-        let nonce = U256::from(5);
-        let send_transaction_tools = &SendTransactionToolsWrapperMock::default()
+        let first_hash = first_signed_transaction.transaction_hash;
+        let second_hash = second_signed_transaction.transaction_hash;
+        let pending_nonce = U256::from(4);
+        //technically, the jason values in those positive responses don't matter, we only check the result and take errors if any came back
+        let rpc_responses = vec![
+            Ok(Value::String((&first_hash.to_string()[2..]).to_string())),
+            Ok(Value::String((&second_hash.to_string()[2..]).to_string())),
+        ];
+        let batch_payables_tools = BatchPayableToolsMock::default()
             .sign_transaction_params(&sign_transaction_params_arc)
-            .sign_transaction_result(Ok(signed_transaction.clone()))
-            .request_new_pending_payable_fingerprint_params(
-                &request_new_pending_payable_fingerprint_params_arc,
-            )
-            .request_new_pending_payable_fingerprint_result(payable_timestamp)
-            .send_raw_transaction_params(&send_raw_transaction_params_arc)
-            .send_raw_transaction_result(Ok(hash));
-        let amount_of_wei = 50_000_000_000_000;
-        let account = make_payable_account_with_recipient_and_balance_and_timestamp_opt(
-            make_wallet("blah123"),
-            amount_of_wei,
+            .sign_transaction_result(Ok(first_signed_transaction.clone()))
+            .sign_transaction_result(Ok(second_signed_transaction.clone()))
+            .batch_wide_timestamp_result(batch_wide_timestamp_expected)
+            .send_new_payable_fingerprint_credentials_params(&new_payable_fingerprint_params_arc)
```

Comment:
The name of the original method may have changed; if so, the mock method name should change too.